src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = true) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  bite(target) {
+    if (Herbivore && !target.hidden) {
+      target.health -= 50;
```

Comment:
[LOGIC ERROR] The condition in the `bite` method is incorrect. It should check if the `target` is an instance of `Herbivore` and not currently hidden. Replace `Herbivore` with `target instanceof Herbivore` to ensure proper functionality.